flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigManager.java

```diff
@@ -157,7 +158,14 @@ public Configuration getObserveConfig(FlinkDeployment deployment) {
             throw new RuntimeException(
                     "Cannot create observe config before first deployment, this indicates a bug.");
         }
-        return getConfig(deployment.getMetadata(), deployedSpec);
+        var conf = getConfig(deployment.getMetadata(), deployedSpec);
+        if (deployment.getSpec().getFlinkConfiguration() != null) {
+            var deployConfig = Configuration.fromMap(deployment.getSpec().getFlinkConfiguration());
+            deployConfig
+                    .getOptional(SAVEPOINT_DIRECTORY)
+                    .ifPresent(dir -> conf.set(SAVEPOINT_DIRECTORY, dir));
```

Comment:
Maybe we could factor this out to a static utility like:
```
applyConfigsFromCurrentSpec(ConfigOption... configs)
```

Reply:
Added.